From 971930ec67e8e2929936307a949d09d38d81a443 Mon Sep 17 00:00:00 2001
From: Xiang Xiaojun <xiangxiaojun@uniontech.com>
Date: Tue, 24 Nov 2020 19:00:52 +0800
Subject: [PATCH 104/171] QPushButton: fix support of style sheet rule for text
 alignment

Fixes: QTBUG-86857
Change-Id: I7e3e2b5323b1e46f572cacfddae20cb6e7882a47
Reviewed-by: Volker Hilsheimer <volker.hilsheimer@qt.io>
(cherry picked from commit 6269438af95bbd988ead08829fa3bc9dc25891e8)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/widgets/styles/qstylesheetstyle.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/widgets/styles/qstylesheetstyle.cpp b/src/widgets/styles/qstylesheetstyle.cpp
index 822e6f895f..72f7ad7455 100644
--- a/src/widgets/styles/qstylesheetstyle.cpp
+++ b/src/widgets/styles/qstylesheetstyle.cpp
@@ -3540,8 +3540,8 @@ void QStyleSheetStyle::drawControl(ControlElement ce, const QStyleOption *opt, Q
                 const uint horizontalAlignMask = Qt::AlignHCenter | Qt::AlignLeft | Qt::AlignRight;
                 const uint verticalAlignMask = Qt::AlignVCenter | Qt::AlignTop | Qt::AlignLeft;
 
-                if (rule.hasPosition() && rule.position()->textAlignment != 0) {
-                    Qt::Alignment textAlignment = rule.position()->textAlignment;
+                const Qt::Alignment textAlignment = rule.position()->textAlignment;
+                if (rule.hasPosition() && textAlignment != 0) {
                     tf |= (textAlignment & verticalAlignMask) ? (textAlignment & verticalAlignMask) : Qt::AlignVCenter;
                     tf |= (textAlignment & horizontalAlignMask) ? (textAlignment & horizontalAlignMask) : Qt::AlignHCenter;
                     if (!styleHint(SH_UnderlineShortcut, button, w))
@@ -3600,6 +3600,8 @@ void QStyleSheetStyle::drawControl(ControlElement ce, const QStyleOption *opt, Q
                         iconRect.translate(pixelMetric(PM_ButtonShiftHorizontal, opt, w),
                                            pixelMetric(PM_ButtonShiftVertical, opt, w));
                     p->drawPixmap(iconRect, pixmap);
+                }else {
+                    tf |= textAlignment;
                 }
 
                 if (button->state & (State_On | State_Sunken))
-- 
2.31.1

